Update added/removed header files in Xcode project

soliton- opened the pull request:
Header files added/removd in commits: c4c292fa3bc0, 43f5644e36e1, 32799739608b, c5b1fa9dfdbb, d954d307bce6, 01f28b12aea5, 63d14217c9c1, 6d77fdc48c6f, 0e85fac29c15, 0b646dec8a13 and 6c5a8e923d52

Pentarctagon commented:
So essentially this is just updating them through xcode so they have the right attributes/are organized correctly compared to what the `add_source_file` script or manually removal does?

soliton- commented:
I removed headers that are gone and added existing ones not in the project. There was also something weird about preferences_list.hpp. Maybe in the wrong section or something. And whatever else Xcode does that I didn’t notice.
